server/controller/src/main/java/ai/starwhale/mlops/domain/dataset/dataloader/DataLoader.java

```diff
@@ -51,14 +51,7 @@ public DataReadLog next(DataReadRequest request) {
 
         dataReadManager.handleConsumerData(consumerId, request.getProcessedData(), session);
 
-        // this lock can be replaced by select fot update in future
-        var lock = new KeyLock<>(String.format("dl-assignment-%s", session.getId()));
-        try {
-            lock.lock();
-            return dataReadManager.assignmentData(consumerId, session);
-        } finally {
-            lock.unlock();
-        }
+        return dataReadManager.assignmentData(consumerId, session);
```

Comment:
It seems that the place that should be blocked is still blocked.
The `select for update` will lock the rows with the same session.id.